### PR TITLE
[14.0][MIG]Migrated account_invoice_refund _reason

### PR DESCRIPTION
**Pull request by Freni-OSI**
Migrated account_invoice_refund _reason from v12 to v14

### CODE REVIEW

**Comment by rvalyi** [comment]
can you rebase please? the offending test has been fixed in https://github.com/OCA/account-invoicing/pull/911

**Review by patrickrwilson — APPROVED** [review]
LGTM

**Review by bodedra — APPROVED** [review]
Functional reviewed from Customer and Vendor side. :+1:

File: account_invoice_refund_reason/wizard/account_move_refund.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        if self.reason_id:
+            self.reason = self.reason_id.name
+
+    def reverse_moves(self):
+        res = super().reverse_moves()
+        mv_obj = self.env["account.move"]
+        context = dict(self._context or {})
+        for inv in mv_obj.browse(context.get("active_ids")):
+            inv.reason_id = self.reason_id.id
+        return res
```
Comment (ivantodorovich) [inline account_invoice_refund_reason/wizard/account_move_refund.py]:
```suggestion
    def reverse_moves(self):
        res = super().reverse_moves()
        self.move_ids.reason_id = self.reason_id
        return res
```

File: account_invoice_refund_reason/views/account_move_view.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            <xpath
+                expr="//page[@id='other_tab_entry']/group/group[1]/field[@name='to_check']"
+                position="after"
+            >
+                <field
+                    name="reason_id"
+                    attrs="{'invisible': [('reason_id', '=', False)]}"
+                    readonly="1"
+                />
+            </xpath>
```
Comment (ivantodorovich) [inline account_invoice_refund_reason/views/account_move_view.xml]:
```suggestion
            <group name="sale_info_group" position="inside">
                <field name="reason_id" attrs="{'invisible': [('reason_id', '=', False)]}" />
            </group>
            <group name="accounting_info_group" position="inside">
                <field name="reason_id" attrs="{'invisible': [('reason_id', '=', False)]}" />
            </group>
```

To keep it simple :)
Note that I removed `readonly="1"` too.  It's better to make it readonly=True in the python model instead

File: account_invoice_refund_reason/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    reason_id = fields.Many2one("account.move.refund.reason", string="Reason")
```
Comment (ivantodorovich) [inline account_invoice_refund_reason/models/account_move.py]:
```suggestion
    reason_id = fields.Many2one("account.move.refund.reason", string="Reason", readonly=True)
```

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Comment by ivantodorovich** [comment]
Thanks a lot @Freni-OSI 
overall it looks good :) could you check my suggestions?

**Comment by tarbib** [comment]
Hello @Freni-OSI any chance you could have a look to @ivantodorovich's suggestions? :)

**Comment by Hardik-OSI** [comment]
Hello @ivantodorovich 
As suggested by suggestion was improved can you please review it now?
Thanks for the suggestion.

File: account_invoice_refund_reason/wizard/account_move_refund.py [hunk 4]
```diff
@@ -16,8 +16,6 @@ def _onchange_reason_id(self):
 
     def reverse_moves(self):
         res = super().reverse_moves()
-        mv_obj = self.env["account.move"]
-        context = dict(self._context or {})
-        for inv in mv_obj.browse(context.get("account_move_id")):
-            inv.reason_id = self.reason_id.id
+        for move in self.move_ids:
+            move.reason_id.id = self.reason_id.id
```
Comment (ivantodorovich) [inline account_invoice_refund_reason/wizard/account_move_refund.py]:
```suggestion
        for move in self.move_ids:
            move.reason_id = self.reason_id
```

or

```suggestion
        self.move_ids.write({"reason_id": self.reason_id.id})
```

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Comment by AaronHForgeFlow** [comment]
Sorry, the module is already on the base branch I am closing this to prevent confusion.

**Pull request closed without merging**